libsolidity/codegen/CompilerUtils.cpp

```diff
@@ -543,17 +542,54 @@ void CompilerUtils::convertType(
 		else if (targetTypeCategory == Type::Category::FixedPoint)
 		{
 			solAssert(
-				stackTypeCategory == Type::Category::Integer || 
+				stackTypeCategory == Type::Category::Integer ||
 				stackTypeCategory == Type::Category::RationalNumber ||
 				stackTypeCategory == Type::Category::FixedPoint,
 				"Invalid conversion to FixedMxNType requested."
 			);
-			//shift all integer bits onto the left side of the fixed type
+
 			FixedPointType const& targetFixedPointType = dynamic_cast<FixedPointType const&>(_targetType);
 			if (auto typeOnStack = dynamic_cast<IntegerType const*>(&_typeOnStack))
-				if (targetFixedPointType.numBits() > typeOnStack->numBits())
+            {
+				if (targetFixedPointType.numBits() > typeOnStack->numBits() + targetFixedPointType.fractionalDigits() * 3)
+				{
 					cleanHigherOrderBits(*typeOnStack);
-			solUnimplemented("Not yet implemented - FixedPointType.");
+				}
+				m_context << u256(pow(bigint(10), targetFixedPointType.fractionalDigits()))
+						  << Instruction::MUL;
+				if (_cleanupNeeded) {
+					cleanHigherOrderBits(targetFixedPointType);
+				}
+            }
+
+            if (auto typeOnStack = dynamic_cast<RationalNumberType const*>(&_typeOnStack))
+			{
+				if (targetFixedPointType.identifier() == (*typeOnStack->fixedPointType()).identifier())
+				{
+					m_context << typeOnStack->literalValue(nullptr);
+				} else
+				{
+					convertType(*typeOnStack, *typeOnStack->fixedPointType());
```

Comment:
This looks like infinite recursion to me...

Reply:
See above